includes/class-wc-emails.php

```diff
@@ -84,17 +84,38 @@ public static function init_transactional_emails() {
 		) );
 
 		foreach ( $email_actions as $action ) {
-			add_action( $action, array( __CLASS__, 'send_transactional_email' ), 10, 10 );
+			add_action( $action, array( __CLASS__, 'queue_transactional_email' ), 10, 10 );
 		}
 	}
 
 	/**
+	 * Queue transactional email via cron so it's not sent in current request.
+	 */
+	public static function queue_transactional_email() {
+		wp_schedule_single_event( time() + 10, 'send_queued_transactional_email', array(
```

Comment:
I think we should probably prefix our events. The actual hook here that gets ran is `send_queued_transactional_email`, where something like `wc_send_queued_transactional_email` would have less of a chance of colliding.